packages/codemirror-blocks/src/ui/PrimitiveList.tsx

```diff
@@ -93,69 +93,139 @@ export const PrimitiveGroup = (props: PrimitiveGroupProps) => {
       {expanded ? (
         <PrimitiveList
           primitives={[...group.flatPrimitivesIter()]}
-          onFocus={onFocus}
-          onKeyDown={onKeyDown}
           selected={selected}
+          selectedPrimitive={selectedPrimitive}
+          toolbarRef={toolbarRef}
+          setSelectedPrimitive={setSelectedPrimitive}
         />
       ) : null}
     </li>
   );
 };
 
 type PrimitiveListProps = {
-  onFocus: BasePrimitiveProps["onFocus"];
-  onKeyDown: BasePrimitiveProps["onKeyDown"];
+  setSelectedPrimitive: (primitive: LanguagePrimitive) => void;
+  selectedPrimitive: LanguagePrimitive | undefined;
```

Comment:
I think it's nicer to do `selectedPrimitive?: LanguagePrimitive` so that the prop can be left off